modules/localstack/src/main/java/org/testcontainers/containers/localstack/LocalStackContainer.java

```diff
@@ -82,20 +82,11 @@ public LocalStackContainer withServices(Service... services) {
         final String address = getContainerIpAddress();
         String ipAddress = address;
         try {
+            // resolve IP address and use that as the endpoint so that path-style access is automatically used for S3
```

Comment:
See [javadocs](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3Builder.html#setPathStyleAccessEnabled-java.lang.Boolean-) - using an IP address for the endpoint ensures that the AWS SDK uses path-style access under the covers, which is the desired effect for the newer version of Localstack.

We use `InetAddress.getByName(...)` to convert the container IP address to an IP regardless of its original form (e.g. could be `localhost`, could already be an IP, could be a hostname). 

The endpoint URL returned by this helper method is strictly for use by the Docker host.